### ISSUE TITLE
Restrict agents from being able to edit their own (previous)threads

### ISSUE THREAD

**Issue by parijathakumar** [osTicket/osTicket#5350]
OsTicket 14.1

I see that there is a way to restrict agents from being able to edit the threads created by other agents. But is there a way to restrict an agent from being able to edit his/her own threads (created by himself or herslf) ?

**Comment by Richiricheh** [comment]
Really like this suggestion

**Comment by JediKev** [comment]
@parijathakumar 

No, there is no way to prevent an Agent from editing his/her own Thread Entry. Reason is because they created the Thread Entry and what if they have mistakes, etc.? They couldn't Edit nor Resend an updated Response to the User. You can always look at the History of Thread Entry edits so you can always see the original Thread Entry.

Cheers.

**Comment by parijathakumar** [comment]
Where to see history of a thread entry ?

**Comment by JediKev** [comment]
@parijathakumar 

You go to an Edited Thread Entry, click the dropdown arrow on the right-hand side of the Entry, select the View History option, and you will see the Original Thread Entry as well as any edits made to the Entry.
<img width="209" alt="Screen Shot 2020-01-24 at 09 12 55" src="https://user-images.githubusercontent.com/11823401/73079546-08aa2a00-3e8a-11ea-9061-54f650f14f6d.png">
<img width="631" alt="Screen Shot 2020-01-24 at 09 13 08" src="https://user-images.githubusercontent.com/11823401/73079550-0b0c8400-3e8a-11ea-918b-e4ad6762d11a.png">
<img width="631" alt="Screen Shot 2020-01-24 at 09 13 15" src="https://user-images.githubusercontent.com/11823401/73079561-0d6ede00-3e8a-11ea-874d-53560e3fa847.png">

Cheers.

**Comment by parijathakumar** [comment]
Thank you. I understood, but we have a different use case here, where agents (external to our organisation) should not edit their previous notes. Otherwise, at least, their edits should also appear more obviously like any other events (field value updates, assignments, etc) without us having to verify the 'history' of each edit. As the OsTicket at present does not have such facility, they are able to change their previous comment and write a complete different comment which is even a complete contradiction to their previous comment. I felt keeping the defaults as such, any of the following two facilities would be helpful in such situations ...

1. There should a role for restricting an agent from being able to edit his/her own thread.
OR
2. The edits should appear more obviously like any other events.

**Comment by JediKev** [comment]
@parijathakumar 

>The edits should appear more obviously like any other events.

They are obvious...? It says `Edited` on the thread entry if it's edited. (look at my screenshots above)

Cheers.

**Comment by parijathakumar** [comment]
True. But what I meant was that it would be even more obvious if the edits appear like an 'event' without having to click the history button on each edit. But now that I come to think of it, this would not be proper, because if an agent has edited lengthy thread, then the event should indicate his entire old thread which would occupy a lot of space and clutter the space. So, if at all this feature request is planned, then I would suggest to consider option-1 ... having a role for restricting agents from being able to edit their own threads.
